src/wp-includes/class-wp-plugin-dependencies.php

```diff
@@ -0,0 +1,884 @@
+			wp_send_json_error( $status );
+		}
+
+		$plugin_file  = self::$plugin_dirnames[ $slug ];
+		$dependencies = self::get_dependencies( $plugin_file );
+
+		if ( empty( $dependencies ) ) {
+			$status['message'] = __( 'The plugin has no required plugins.' );
+			wp_send_json_success( $status );
+		}
+
+		$inactive_dependencies = array();
+		foreach ( $dependencies as $dependency ) {
+			if ( false === self::$plugin_dirnames[ $dependency ] || is_plugin_inactive( self::$plugin_dirnames[ $dependency ] ) ) {
+				$inactive_dependencies[] = $dependency;
+			}
+		}
+
+		if ( ! empty( $inactive_dependencies ) ) {
+			$inactive_dependency_names = array_map(
+				function ( $dependency ) {
+					if ( isset( self::$dependency_api_data[ $dependency ]['Name'] ) ) {
+						$inactive_dependency_name = self::$dependency_api_data[ $dependency ]['Name'];
+					} else {
+						$inactive_dependency_name = $dependency;
+					}
+					return $inactive_dependency_name;
+				},
+				$inactive_dependencies
+			);
+
+			$status['errorCode']    = 'inactive_dependencies';
+			$status['errorMessage'] = sprintf(
+				/* translators: %s: A list of inactive dependency plugin names. */
+				__( 'The following plugins must be activated first: %s.' ),
+				implode( ', ', $inactive_dependency_names )
+			);
+			$status['errorData'] = array_combine( $inactive_dependencies, $inactive_dependency_names );
+
+			wp_send_json_error( $status );
+		}
+
+		$status['message'] = __( 'All required plugins are installed and activated.' );
+		wp_send_json_success( $status );
+	}
+
+	/**
+	 * Stores the result of 'get_plugins()'.
+	 *
+	 * @since 6.4.0
+	 */
+	protected static function get_plugins() {
+		if ( ! function_exists( 'get_plugins' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/plugin.php';
+		}
+		self::$plugins = get_plugins();
+	}
+
+	/**
+	 * Reads and stores dependency slugs from a plugin's 'Requires Plugins' header.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @global WP_Filesystem_Base $wp_filesystem WordPress filesystem subclass.
+	 */
+	protected static function read_dependencies_from_plugin_headers() {
+		global $wp_filesystem;
+
+		if ( ! $wp_filesystem ) {
+			require_once ABSPATH . '/wp-admin/includes/file.php';
+			WP_Filesystem();
+		}
+
+		self::get_plugins();
+		$plugins_dir     = trailingslashit( $wp_filesystem->wp_plugins_dir() );
+		$default_headers = array( 'RequiresPlugins' => 'Requires Plugins' );
+
+		foreach ( array_keys( self::$plugins ) as $plugin ) {
+			$header = get_file_data( $plugins_dir . $plugin, $default_headers, 'plugin' );
+
+			if ( '' === $header['RequiresPlugins'] ) {
+				continue;
+			}
+
+			$dependency_slugs              = self::sanitize_dependency_slugs( $header['RequiresPlugins'] );
+			self::$dependencies[ $plugin ] = $dependency_slugs;
+			self::$dependency_slugs        = array_merge( self::$dependency_slugs, $dependency_slugs );
+		}
+
+		$dependent_keys         = array_keys( self::$dependencies );
```

Comment:
Let's add protection for single file plugins.
```suggestion
		$dependent_keys = array();
		foreach ( array_keys( self::$dependencies ) as $dependency ) {
			$dependent_keys[] = str_contains( $dependency , '/' ) ? dirname( $dependency ) : $dependency;
		}
```